lib/manageiq_foreman/lib/manageiq_foreman/connection.rb

```diff
@@ -1,22 +1,13 @@
 module ManageiqForeman
   class Connection
-    CLASSES = {
-      :config_templates  => ForemanApi::Resources::ConfigTemplate,
-      :home              => ForemanApi::Resources::Home,
-      :hostgroups        => ForemanApi::Resources::Hostgroup,
-      :hosts             => ForemanApi::Resources::Host,
-      :media             => ForemanApi::Resources::Medium,
-      :operating_systems => ForemanApi::Resources::OperatingSystem,
-      :ptables           => ForemanApi::Resources::Ptable,
-      :subnets           => ForemanApi::Resources::Subnet,
-      :locations         => ForemanApi::Resources::Location,
-      :organizations     => ForemanApi::Resources::Organization,
-    }
-
     attr_accessor :connection_attrs
 
-    def initialize(connection_attrs)
-      @connection_attrs = connection_attrs
+    def initialize(attrs)
+      self.connection_attrs = attrs.dup
+      connection_attrs[:uri] = connection_attrs.delete(:base_url)
+      connection_attrs[:api_version] ||= 2
+      connection_attrs[:apidoc_cache_dir] ||= defined?(Rails) ? Rails.root.join("tmp/foreman").to_s : "/tmp/foreman"
```

Comment:
In the non-Rails case, wondering if you should use [Dir.tmpdir](http://ruby-doc.org/stdlib-2.0.0/libdoc/tmpdir/rdoc/Dir.html#method-c-tmpdir), in case /tmp does not exist or is different?